### PR TITLE
Implemented SpawnExperienceOrb and SpawnPainting

### PR DESCRIPTION
**Pull request by roxxel**

### CODE REVIEW

File: Obsidian.API/_Types/Painting.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+namespace Obsidian.API
+{
+    public class Painting
+    {
+
+        public readonly short Id;
+        public readonly byte X;
+        public readonly byte Y;
+        public readonly byte Width;
+        public readonly byte Height;
```
Comment (Seb-stian) [inline Obsidian.API/_Types/Painting.cs]: These could be properties.

_Since it's not a lot of data, `Painting` could even be a struct?_

File: Obsidian/WorldData/World.cs [hunk 2]
```diff
@@ -483,5 +483,22 @@ internal bool TryAddEntity(Entity entity)
...
+                Position = position,
+                EntityId = 2 
+                // 2 = Experience Orb Source: https://minecraft.gamepedia.com/Java_Edition_data_values/Pre-flattening/Entity_IDs
+            });
+        }
+
+        public async Task SpawnPainting(Position position, Painting painting, PaintingDirection direction, Guid uuid = default)
+        {
+            if (uuid == Guid.Empty) uuid = Guid.NewGuid();
+            await this.Server.BroadcastPacketAsync(new SpawnPainting(9, uuid, painting.Id, position, direction));
```
Comment (Seb-stian) [inline Obsidian/WorldData/World.cs]: I wonder where does the magic `9` come from?

File: Obsidian/Net/Packets/Play/Clientbound/SpawnPainting.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        public Guid UUID;
+
+        [Field(2), VarLength]
+        public int Motive;
+
+        [Field(3)]
+        public Position Position;
+
+        [Field(4), ActualType(typeof(byte))]
+        public PaintingDirection Direction;
```
Comment (Seb-stian) [inline Obsidian/Net/Packets/Play/Clientbound/SpawnPainting.cs]: Also could be properties.

**Review by Seb-stian — APPROVED** [review]
Looks really good 👍

File: Obsidian.API/_Interfaces/IWorld.cs [hunk 4]
```diff
@@ -7,5 +10,8 @@ public interface IWorld
 
         public long Time { get; }
         public Gamemode GameType { get; }
+
+        public Task SpawnExperienceOrb(PositionF position, short count);
```
Comment (Craftplacer) [inline Obsidian.API/_Interfaces/IWorld.cs]: What does `count` mean? The variable name is unclear and misses XML docs for clarification.

File: Obsidian/WorldData/World.cs (hunk 2 above)
Comment (Craftplacer) [reply Obsidian/WorldData/World.cs]: > I wonder where does the magic `9` come from?

The comment above seems related, https://minecraft.gamepedia.com/Java_Edition_data_values/Pre-flattening/Entity_IDs
says that 9 is the Entity ID for paintings.

**Review by Craftplacer — APPROVED** [review]

File: Obsidian.API/_Interfaces/IWorld.cs (hunk 4 above)
Comment (Seb-stian) [reply Obsidian.API/_Interfaces/IWorld.cs]: I'm pretty sure that it's the experience orb count. Maybe the method should be renamed to `SpawnExperienceOrbs`.

Comment (Naamloos) [reply Obsidian.API/_Interfaces/IWorld.cs]: A rename of the method to more clearly define that it's multiple orbs would be nice, yes.

File: Obsidian.API/_Types/Painting.cs (hunk 1 above)
Comment (Naamloos) [reply Obsidian.API/_Types/Painting.cs]: +1 on that

File: Obsidian/WorldData/World.cs (hunk 2 above)
Comment (Naamloos) [reply Obsidian/WorldData/World.cs]: If the entity ID for paintings is _always_ 9, it should be in `SpawnPainting.cs` and setting a different entity ID shouldn't even be an option in the constructor. I'd say either hard code it in the `SpawnPainting` constructor, on one of it's properties or look if there's any loaded data for entity IDs in the registry.

**Review by Naamloos — CHANGES_REQUESTED** [review]
I recommend looking at the given feedback and make changes accordingly. It looks good overall, but some minor changes can be made. Once that is done, LGTM and merge ready.

but first, I'll mark this as a request for changes

File: Obsidian/WorldData/World.cs (hunk 2 above)
Comment (Naamloos) [reply Obsidian/WorldData/World.cs]: Addition to that, if the entity id for paintings is static then it's kinda weird that mojang defines it in the packets. hmm

File: Obsidian.API/_Types/Painting.cs [hunk 5]
```diff
@@ -6,14 +6,14 @@
...
-    public class Painting
+    public struct Painting
     {
 
-        public readonly short Id;
-        public readonly byte X;
-        public readonly byte Y;
-        public readonly byte Width;
-        public readonly byte Height;
+        public short Id { get; private set; }
```
Comment (Craftplacer) [inline Obsidian.API/_Types/Painting.cs]: Properties can be set in the constructor no matter if the `set` is missing, I would suggest to remove it.

**Review by Craftplacer — APPROVED** [review]

File: Obsidian.API/_Types/Painting.cs (hunk 5 above)
Comment (Seb-stian) [reply Obsidian.API/_Types/Painting.cs]: And then even make the struct `readonly`.

File: Obsidian/Net/Packets/Play/Clientbound/SpawnExperienceOrb.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+        [Field(1), Absolute]
+        public PositionF Position { get; private set; }
+
+        [Field(2)]
+        public short Count { get; private set; }
+
+        public int Id => 0x01;
+
+
```
Comment (Seb-stian) [inline Obsidian/Net/Packets/Play/Clientbound/SpawnExperienceOrb.cs:25]: There should be an empty constructor (at least a private one). I don't think the project will even compile without it. (generators)

**Review by Seb-stian — CHANGES_REQUESTED** [review]
Needs changes, see my SpawnExperienceOrb.cs comment.

**Review by Seb-stian — APPROVED** [review]

**Comment by Seb-stian** [comment]
With approval from @Naamloos I would merge.

**Review by Naamloos — APPROVED** [review]
New changes look good to me, nice work 👌👍

**Pull request merged**